ledger/src/shred.rs

```diff
@@ -841,6 +840,17 @@ where
                 stats.shred_flags_bad_deserialize += 1;
                 return true;
             };
+
+            if shred_flags.contains(ShredFlags::DATA_COMPLETE_SHRED)
+                && index != fec_set_index + DATA_SHREDS_PER_FEC_BLOCK as u32 - 1
+            {
+                stats.unexpected_data_complete_shred += 1;
+
+                if discard_unexpected_data_complete_shreds(slot) {
```

Comment:
```suggestion
                if enforce_fixed_fec_set(slot) && discard_unexpected_data_complete_shreds(slot) {
```
Since the math above with `DATA_SHREDS_PER_FEC_BLOCK` will only work once we have fixed fec sets